gobblin-modules/gobblin-kafka-09/src/test/java/org/apache/gobblin/runtime/DagManagementDagActionStoreChangeMonitorTest.java

```diff
@@ -0,0 +1,149 @@
+ * events stored in a {@link org.apache.gobblin.kafka.client.KafkaConsumerRecord}. The
+ * processMessage(DecodeableKafkaRecord message) function should be able to gracefully process a variety of message
+ * types, even with undesired formats, without throwing exceptions.
+ */
+@Slf4j
+public class DagManagementDagActionStoreChangeMonitorTest {
+  public static final String TOPIC = DagActionStoreChangeEvent.class.getSimpleName();
+  private final int PARTITION = 1;
+  private final int OFFSET = 1;
+  private final String FLOW_GROUP = "flowGroup";
+  private final String FLOW_NAME = "flowName";
+  private final long FLOW_EXECUTION_ID = 123L;
+  private final String JOB_NAME = "jobName";
+  private MockDagManagementDagActionStoreChangeMonitor mockDagManagementDagActionStoreChangeMonitor;
+  private int txidCounter = 0;
+
+  private static final DagActionReminderScheduler dagActionReminderScheduler = mock(DagActionReminderScheduler.class);
+
+  /**
+   * Note: The class methods are wrapped in a test specific method because the original methods are package protected
+   * and cannot be accessed by this class.
+   */
+  static class MockDagManagementDagActionStoreChangeMonitor extends DagManagementDagActionStoreChangeMonitor {
+
+    public MockDagManagementDagActionStoreChangeMonitor(Config config, int numThreads, boolean isMultiActiveSchedulerEnabled) {
+      super(config, numThreads, mock(FlowCatalog.class), mock(Orchestrator.class), mock(DagManagementStateStore.class),
+          isMultiActiveSchedulerEnabled, mock(DagManagement.class), dagActionReminderScheduler);
+    }
+    protected void processMessageForTest(DecodeableKafkaRecord<String, DagActionStoreChangeEvent> record) {
+      super.processMessage(record);
+    }
+  }
+
+  MockDagManagementDagActionStoreChangeMonitor createMockDagManagementDagActionStoreChangeMonitor() {
+    Config config = ConfigFactory.empty().withValue(ConfigurationKeys.KAFKA_BROKERS, ConfigValueFactory.fromAnyRef("localhost:0000"))
+        .withValue(Kafka09ConsumerClient.GOBBLIN_CONFIG_VALUE_DESERIALIZER_CLASS_KEY, ConfigValueFactory.fromAnyRef("org.apache.kafka.common.serialization.ByteArrayDeserializer"))
+        .withValue(ConfigurationKeys.STATE_STORE_ROOT_DIR_KEY, ConfigValueFactory.fromAnyRef("/tmp/fakeStateStore"))
+        .withValue("zookeeper.connect", ConfigValueFactory.fromAnyRef("localhost:2121"));
+    return new MockDagManagementDagActionStoreChangeMonitor(config, 5, true);
+  }
+
+  // Called at start of every test so the count of each method being called is reset to 0
+  @BeforeMethod
+  public void setupMockMonitor() {
+     mockDagManagementDagActionStoreChangeMonitor = createMockDagManagementDagActionStoreChangeMonitor();
+  }
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    doNothing().when(dagActionReminderScheduler).unscheduleReminderJob(any());
+
+  }
+
+  /**
+   * Tests process message with a DELETE type message which should be ignored regardless of the flow information.
```

Comment:
is this desc accurate? the delete is not ignored anymore

Reply:
bad copy paste